app.js

```diff
@@ -180,9 +180,58 @@ class FilterBox1 {
   }
 }
 
-const filterBox1 = new FilterBox1();
+const filterBox1 = new FilterBoxOpenAndClose();
 
-class FilterBox2 {}
+class FilterByType {
+  constructor() {
+    this.onlineCheckbox = document.querySelector("#onlineChallengeCheckbox");
+    this.onsiteCheckbox = document.querySelector("#onsiteChallengeCheckbox");
+
+    this.onlineCheckbox.addEventListener("change", () => {
+      this.filterChallengesByType();
+    });
+
+    this.onsiteCheckbox.addEventListener("change", () => {
+      this.filterChallengesByType();
+    });
+  }
+
+  filterChallengesByType() {
+    const onlineChecked = this.onlineCheckbox.checked;
+    const onsiteChecked = this.onsiteCheckbox.checked;
+
+    const challengesContainer = document.querySelector("#ourChallenges");
+    const api = new APIadapter();
+
+    api.fetchAllChallenges().then((challenges) => {
+      let filteredChallenges;
+
+      if (!onlineChecked && !onsiteChecked) {
+        // No checkboxes are checked, show all challenges
+        filteredChallenges = challenges;
+      } else {
+        // Filter challenges
+        filteredChallenges = challenges.filter((challenge) => {
+          return (
+            (onlineChecked && challenge.data.type === "online") ||
+            (onsiteChecked && challenge.data.type === "onsite")
+          );
+        });
+      }
+
+      // Clear challenges
+      challengesContainer.innerHTML = "";
+
+      // Render challenges
+      filteredChallenges.forEach((challenge) => {
+        const element = challenge.render();
+        challengesContainer.appendChild(element);
+      });
+    });
+  }
+}
+
+const filterBox2 = new FilterByType();
```

Comment:
This variable is not used anywhere. When you create a class instance but don't use it anywhere, that's a red flag. It tells me that either the class doesn't do anything, or (like here) it interacts with the outside world directly, which is a seapration of concerns issue. It means that the code isn't very object oriented (in this regard) even if implemented as a class.